src/pybind/mgr/dashboard/frontend/src/app/shared/services/favicon.service.ts

```diff
@@ -0,0 +1,78 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable, OnDestroy } from '@angular/core';
+
+import { Subscription } from 'rxjs';
+
+import { SummaryService } from './summary.service';
+
+@Injectable()
+export class FaviconService implements OnDestroy {
+  sub: Subscription;
+  oldStatus: string;
+  url: string;
+
+  constructor(
+    @Inject(DOCUMENT) private document: HTMLDocument,
+    private summaryService: SummaryService
+  ) {}
+
+  init() {
+    this.url = this.document.getElementById('cdFavicon')?.getAttribute('href');
+
+    this.sub = this.summaryService.subscribe((summary) => {
+      this.changeIcon(summary.health_status);
+    });
+  }
+
+  changeIcon(status?: string) {
+    if (status === this.oldStatus) {
+      return;
+    }
+
+    this.oldStatus = status;
+
+    const favicon = this.document.getElementById('cdFavicon');
+    const faviconSize = 16;
+
+    const canvas = this.document.createElement('canvas');
+    canvas.width = faviconSize;
+    canvas.height = faviconSize;
+
+    const context = canvas.getContext('2d');
+    const img = this.document.createElement('img');
+    img.src = this.url;
+
+    img.onload = () => {
+      // Draw Original Favicon as Background
+      context.drawImage(img, 0, 0, faviconSize, faviconSize);
+
+      // Draw Notification Circle
+      context.beginPath();
+      context.arc(canvas.width - faviconSize / 4, faviconSize / 4, faviconSize / 4, 0, 2 * Math.PI);
+
+      let color = 'transparent';
+      switch (status) {
+        case 'HEALTH_OK':
+          color = '#00bb00';
+          break;
+        case 'HEALTH_WARN':
+          color = '#ffa500';
+          break;
+        case 'HEALTH_ERR':
+          color = '#ff0000';
+          break;
```

Comment:
Can colors be extracted to a constants file, just no to have the same colors repeated (e.g.: HealthPipe, ...)?

Reply:
sure, I can do that.

Reply:
Thanks!